### DISCUSSION TITLE
On Android stop GPS Background Service when app exits

### DISCUSSION THREAD

**Discussion by bb281** [shinyorg/shiny#893] [Q&A] [answered]
On iOS, it appears the GPS Background Service stops when the main app closes. This is what I want to do.
However, on Android, it appears that the GPS Background Service stays up even after the app exits.

Any tips/suggestions on Android to stop the background service when the app exits?

**Comment by aritchie** [comment, marked as answer]
This is easy to accomplish, but I'm not sure why you are registering the background mode if you don't want it.  You can IGpsManager.WhenReading() with StartListener/StopListener when the app enters/exists.

**Reply by bb281** [reply]
Thanks for the quick response. To clarify, I want the behavior to emulate the iOS when going to the background. For example, when the user puts the app in the background by switching to another app, I want to continue to retrieve the GPS. I'm assuming that I have to be in background mode to accomplish that. Thus, I'm registering the background mode.
The issue I have on the Android side is when the user closes the main application, the background service continues to run and I'd like to close it like in iOS. I've ran the sample GPS.sln solution locally and it has the same behavior.

I'm not sure if what I'm describing above is possible on Android. I'm creating a start/stop GPS background in the app to give the user some control. Thanks for any feedback/input.

**Reply by aritchie** [reply]
You say in your first statement you want to "continue to retrieve GPS".  In the next statement, you are saying you want it to close on Android.  I'm confused as to what you want.  

If it is stopping, when your app is going to the background call GpsManager.StopListener() as I mentioned in my first answer.

If you want it to keep running, make sure you call GpsManager.StartListener(GpsRequest.Realtime(true)) or equivalent to your settings.
